### PR TITLE
Initial commit of "Knative Migration-To-Core Process"

### PR DESCRIPTION
**Pull request by evankanderson**
From here:  https://docs.google.com/document/d/1Mgg__nCNVWj0jUdJMC15JjtPpPFyGO21uSdAUvPGU-E/edit

### CODE REVIEW

File: mechanics/MATURITY-LEVELS.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+## Usable
+
+A usable project has reached the point where it is releasing executable
+artifacts which are in use by at least two end users (preferably different
+companies, rather than installations within the same company). Additionally, a
+usable project meets certain other criteria (defined below) which allows the
+Knative community to have confidence in the project and willingness to invest
+further.
+
+Additional requirements beyond "initiating":
```
Comment (lionelvillard) [inline mechanics/MATURITY-LEVELS.md:103]: Should usable projects be in the knative-sandbox org?

**Comment by evankanderson** [comment]
I believe this is ready to merge; I'd cleaned up the broken links from the export but forgot to remove the large, annoying header.

**Comment by evankanderson** [comment]
/assign @pmorie 

Paul helped a lot with shepherding this process, I'm transcribing the (shared) doc to Markdown to make it official.

/assign @ronavn

Since Ron was interested in the "scope of Knative" question that we discussed in the functions WG retrospective.

**Comment by evankanderson** [comment]
/assign @maximilien 

Since he had a bunch of insights on the Cloud Foundry process.

**Comment by vaikas** [comment]
@evankanderson what's the status of this PR? Looks like you were waiting on some comments from folks, so just giving it a bump.
@ronavn @maximilien @pmorie for comments / thoughts.

**Comment by evankanderson** [comment]
I think most of the comments had been resolved; I'm not sure that trademark needs to weigh in, since this is about repo organization and not mark usage (both `knative` and `knative-sandbox` use the mark).

**Comment by evankanderson** [comment]
Updated link to point to [the Knative-owned copy of the document](https://docs.google.com/document/d/1Mgg__nCNVWj0jUdJMC15JjtPpPFyGO21uSdAUvPGU-E/edit).

**Review by vaikas — APPROVED** [review]
/lgtm
Looks great, we can tune as we deem it necessary in the future.

File: mechanics/MATURITY-LEVELS.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+appropriate. Note that projects may not map 1:1 with repos, and projects may
+even be hierarchical (for example, the "eventing" project includes "eventing
+API", "eventing delivery" and "eventing sources", each of which has artifacts in
+multiple repositories).
+
+Note that this document intentionally refers to the process of selecting
+implemented interfaces from sandbox which should be part of core as
+"migration-to-core" rather than "incubation" or "promotion" – while it may be
+personally satisfying to have a component be part of the core, components which
+are not in the core may be equally valuable and production-worthy.
```
Comment (csantanapr) [inline mechanics/MATURITY-LEVELS.md:17]: what about `production ready` instead?

**Pull request merged**